libs/community/langchain_community/vectorstores/azuresearch.py

```diff
@@ -584,6 +584,29 @@ def semantic_hybrid_search_with_score(
         )
         return [(doc, score) for doc, score, _ in docs_and_scores]
 
+    def semantic_hybrid_search_with_rerank(
```

Comment:
what are your opinions on modifying `semantic_hybrid_search_with_score` to:

- accept `score_type: Literal["score", "reranker_score"] = "score"` to control which score is used in `docs_and_scores`;
- check for `score_threshold` as you do here.

Reply:
Makes sense to me -- It would also make the vector store retriever more versatile. I've just appended one commit.